### PR TITLE
dotnet 7 compatibility things

### PR DESCRIPTION
**Pull request by WebFreak001**
I don't know, I couldn't get the project to build without errors, which I all fixed here.

dotnet told me PackageLicenseUrl is deprecated/removed

### CODE REVIEW

File: osu.Framework.Font/Graphics/Sprites/KaraokeSpriteText_DrawNode.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ protected override long GetDrawVersion()
             return base.GetDrawVersion();
         }
 
-        public List<DrawNode> Children
+        public List<DrawNode>? Children
```
Comment (andy840119) [inline osu.Framework.Font/Graphics/Sprites/KaraokeSpriteText_DrawNode.cs:69]: ~~Not very sure about should let it become nullable?~~
~~Because I remember `InternalChildren` in the `CompositeDrawable` will not be nullable if apply the `NRT`. so the same things should apply in the draw node i guess?~~

[Update]
peppy said it's nullable, so it's nullable 
https://github.com/ppy/osu-framework/commit/e9f08ad8874a86b08dcea54c63f0a56acf993435

File: osu.Framework.Font/osu.Framework.Font.csproj [hunk 2]
```diff
@@ -32,7 +31,8 @@
     <PackageReference Include="ppy.osu.Framework" Version="2023.228.0" />
   </ItemGroup>
   <ItemGroup>
-    <EmbeddedResource Include="Resources\**\*" />
+    <!-- commented out since the shaders use a .fs extension, which already seems to be loaded, maybe because of F#, causing build errors... -->
+    <!-- <EmbeddedResource Include="Resources\**\*" /> -->
```
Comment (andy840119) [inline osu.Framework.Font/osu.Framework.Font.csproj:35]: I think should revert this change because it's out of the topic?

File: osu.Framework.Font/osu.Framework.Font.csproj [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿<Project Sdk="Microsoft.NET.Sdk">
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (andy840119) [inline osu.Framework.Font/osu.Framework.Font.csproj:1]: include this one.

File: osu.Framework.Font/Graphics/Containers/MaskingContainer.cs [hunk 4]
```diff
@@ -74,7 +74,7 @@ private static RectangleI generateMasking(RectangleI source, Edges edges)
             int getExtendSize(Edges flag)
             {
                 const int extend_size = 1000;
-                return edges.HasFlag(flag) ? 0 : extend_size;
+                return (edges & flag) == flag ? 0 : extend_size;
```
Comment (andy840119) [inline osu.Framework.Font/Graphics/Containers/MaskingContainer.cs:77]: Not very sure show use `HasFlagFast` follow some part in the osu-framework?

File: osu.Framework.Font/Graphics/Sprites/KaraokeSpriteText.cs [hunk 5]
```diff
@@ -411,7 +411,7 @@ protected override bool OnInvalidate(Invalidation invalidation, InvalidationSour
     {
         var result = base.OnInvalidate(invalidation, source);
 
-        if (!invalidation.HasFlag(Invalidation.Layout))
```
Comment (andy840119) [inline osu.Framework.Font/Graphics/Sprites/KaraokeSpriteText.cs:414]: `HasFlagFast` i guess?

**Review by andy840119 — APPROVED** [review]
Thanks for the contribution!
Most of part LGTM.

File: osu.Framework.Font/osu.Framework.Font.csproj (hunk 3 above)
Comment (WebFreak001) [reply osu.Framework.Font/osu.Framework.Font.csproj:1]: I don't know what has changed here

File: osu.Framework.Font/osu.Framework.Font.csproj (hunk 2 above)
Comment (WebFreak001) [reply osu.Framework.Font/osu.Framework.Font.csproj:35]: well it doesn't compile if the EmbeddedResources are in the file on my machine (arch)

File: osu.Framework.Font/Graphics/Containers/MaskingContainer.cs (hunk 4 above)
Comment (WebFreak001) [reply osu.Framework.Font/Graphics/Containers/MaskingContainer.cs:77]: I couldn't figure out how to use HasFlagFast from the error message, I tried just doing edges.HasFlagFast(flag), but that didn't exist on the enum, so I just manually done bit operations here since it didn't seem to exist.

File: osu.Framework.Font/osu.Framework.Font.csproj (hunk 3 above)
Comment (andy840119) [reply osu.Framework.Font/osu.Framework.Font.csproj:1]: 🤔
Which IDE are you using?

Comment (WebFreak001) [reply osu.Framework.Font/osu.Framework.Font.csproj:1]: vscode, but the C# extension didn't work with my folder layout, so it was pretty much just a text editor with no C# functionality.

Comment (andy840119) [reply osu.Framework.Font/osu.Framework.Font.csproj:1]: Guess it's time to try the rider
PM me if you are interesting

File: osu.Framework.Font/Graphics/Containers/MaskingContainer.cs (hunk 4 above)
Comment (andy840119) [reply osu.Framework.Font/Graphics/Containers/MaskingContainer.cs:77]: HasFlagFast is a extension method in the osu-framework

**Comment by andy840119** [comment]
Also, @WebFreak001 could you help me to rebase this PR into the latest master branch?
And make one commit for one kind of change,thanks 🙏

**Review by andy840119 — APPROVED** [review]
LGTM
